src/main/java/net/sf/mzmine/modules/peaklistmethods/filtering/duplicatefilter/DuplicateFilterTask.java

```diff
@@ -167,8 +167,9 @@ private PeakList filterDuplicatePeakListRows(final PeakList origPeakList, final
     for (int firstRowIndex = 0; !isCanceled() && firstRowIndex < rowCount; firstRowIndex++) {
 
       final PeakListRow mainRow = peakListRows[firstRowIndex];
+
       if (mainRow != null) {
-        // need to copy to not alter the old peakList
+        // copy first row
         PeakListRow firstRow = copyRow(mainRow);

```

Comment:
We need to copy each firstRow to be inserted later in the resulting peak list. 
As we can have multiple duplicate second rows, it is best to copy the row before the second loop. Then all features with a higher PeakStatus are exchanged.